api/openapi/my_api.yaml

```diff
@@ -225,263 +246,194 @@ components:
+          example: "38-1"
+        inputs:
+          type: object
+          items:
+            $ref: '#/components/schemas/Settings'
+        return:
+          type: object
+          items:
+            $ref: '#/components/schemas/Settings'
+        cards:
           type: array
           items:
-            type: string
-            description: "An array of container names upon which the plugin in question depends. For example, 'txscience/pds-fhir-example:0.2.0'. Mapper plugins must indicate a dependency on one or more FHIR plugins, but dependencies are discouraged otherwise in order to maintain the decoupling and reuse of plugins. Every plugin of type 'm' and 'mD' must have at least one 'f' or 'fD' plugin listed in pluginDependencies, otherwise return 500."
+            $ref: '#/components/schemas/Card'
+        specs:
+          type: array
+          items:
+            $ref: '#/components/schemas/Spec'
+        timeout:
+          type: integer
+          description: "Number of milliseconds until time out. Factory system default should be 120000 (2 minutes) for RUO applictions, and only 500ms for clinical."
+        fhir_piid:
+          type: string
+          example: "pdspi-fhir-example"
+          description: "Uniquely identifies the enabled and installed plugin to use for mapper"
+        mapper_piid:
+          type: string
+          example: "pdspi-mapper-example"
+          description: "Uniquely identifies the enabled and installed plugin to use for fhir"
+    Settings:
+      type: object
+      properties:
         pluginSelectors:
-          description: "Maps the piid to a set of selector values. Optionally call the /selectors endpoint first to get the full list of available selectors and their legal values. For example, two selectors might be 1. drug type and 2. indication. If the client selects values for the selector that matches all piid selector values, then this plugin can provide guidance. Potentially, a decision tree could be implemented here by prefacing each selectorarray item with logic, but for this API version we will instead always AND all the selector array elements. For example, 'if guidance is requried for drug _x_ AND indication _y_ then use plugin _z_'."
+          description: "Maps the piid to a set of selector values."
           type: array
           items:
-            $ref: '#/components/schemas/Selector'
-        pluginParameterDefaults:
+            $ref: '#/components/schemas/PluginSelector'
+        modelParameters:
           description: "Default, typed values for any parameters that might be required by the plugin"
           type: array
           items:
             $ref: '#/components/schemas/PluginParameter'
-        requiredPatientVariables:
+        patientVariables:
           type: array
           items:
-            $ref: '#/components/schemas/RequiredPatientVariable'
-    PluginParameter:
```

Comment:
Here's where I"m getting lost. Maybe we should first make changes to the fields, and make changes to the order in a second commit, for the purpose of making the review easier? Was there anything in particular you had questions about in this section?